src/PyCall.jl

```diff
@@ -576,6 +636,25 @@ pycall(o::Union{PyObject,PyPtr}, ::Type{PyAny}, args...; kwargs...) =
 @compat (o::PyObject)(args...; kws...) = pycall(o, PyAny, args...; kws...)
 @compat (::Type{PyAny})(o::PyObject) = convert(PyAny, o)
 
+pytypeof(o::PyObject) = PyObject(@pycheckn ccall(@pysym(:PyObject_Type), PyPtr, (PyPtr,), o))
+
+
+"""
+    @pycall func(args...)::T
+
+Convenience macro which turns `func(args...)::T` into pycall(func, T, args...)
+"""
+macro pycall(ex)
+    @assert (isa(ex,Expr) && (ex.head==:(::)) 
```

Comment:
This should just be a standard exception (probably `ArgumentError`), not an assertion failure.  Assertion failures should only be used when failure would indicate a bug in PyCall itself.